docs/Using Fleet/MDM-macOS-setup.md

```diff
@@ -16,7 +16,116 @@ MacOS setup features require connecting Fleet to Apple Business Manager (ABM). L
 
 ## End user authentication
 
-> This feature is currently in development.
+Using Fleet, you can require end users to authenticate with your identity provider (IdP) and agree to an end user license agreement (EULA) before they can use their new Mac.
+
+To require end user authentication, we will do the following steps:
+
+1. Connect Fleet to your IdP
+2. Upload a EULA to Fleet
+3. Enable end user authentication
+
+### Step 1: connect Fleet to your IdP
+
+In Fleet, you can connect Fleet to your IdP using the Fleet UI or fleetctl command-line tool.
+
+The Fleet UI method is a good start if you're just getting familiar with Fleet.
+
+The fleetctl CLI method enables managing your IdP credentials in a Git repository. This way you can enforce code review and benefit from Git's change history.
+
+Fleet UI:
+
+1. Head to the **Settings > Integrations > Automatic enrollment** page.
+
+2. Under **End user authentication**, enter your IdP credentials and select **Save**.
+
+fleetctl CLI:
+
+1. Create `fleet-config.yaml` file:
+
+```yaml
+apiVersion: v1
+kind: config
+spec:
+  mdm:
+    end_user_authentication:
+      identity_provider_name: "Okta"
+      entity_id: 123
+      issuer_url: "https://example.com"
+      metadata_url: "https://example.com"
+  ...
+```
+
+2. Add an `mdm.end_user_authentication` key to your YAML document. This key accepts the above keys for your IdP credentials. 
+
+3. Run the fleetctl `apply -f fleet-config.yml` command to add your IdP credentials.
+
+4. Confirm that your IdP credentials were saved by running `fleetctl get config`.
+
+Learn more about "No team" configuration options [here](./configuration-files/README.md#organization-settings).
+
+### Step 2: upload a EULA to Fleet
+
+1. Head to the **Settings > Integrations > Automatic enrollment** page.
+
+2. Under **End user license agreement (EULA)**, select **Upload** and choose your EULA.
+
+### Step 3: enable end user authentication
+
+You can enable end user authentication using the Fleet UI or fleetctl command-line tool.
+
+Fleet UI:
+
+1. Head to the **Controls > macOS settings > macOS setup > End user authentication** page.
+
+2. Choose which team you want to enable end user authentication for to to by selecting the desired team in the teams dropdown in the upper left corner.
+
+3. Select the **On** checkbox and select **Save**.
+
+fleetctl CLI: 
+
+1. Choose which team you want to enable end user authentication on.
+
+In this example, we'll enable end user authentication on the "Workstations (canary)" team so that the authentication is only required for hosts that automatically enroll to this team.
```

Comment:
Indenting a few spaces will align this paragraph with the 1st list item:
```suggestion
   In this example, we'll enable end user authentication on the "Workstations (canary)" team so that the authentication is only required for hosts that automatically enroll to this team.
```